pkg/providers/aws/provider_postgres.go

```diff
@@ -674,3 +696,55 @@ func (p *PostgresProvider) setPostgresServiceMaintenanceMetric(ctx context.Conte
 
 	return nil
 }
+
+// CreateRDSAvailabilityAlert Call this when we create the ElastiCache instance to create an
+// alert to watch for the availability of the instance
+func (p *PostgresProvider) CreateRDSAvailabilityAlert(ctx context.Context, cr *v1alpha1.Postgres, instanceID string, clusterID string) error {
+	alertRuleName := fmt.Sprintf("cro-aws-rds-%s", instanceID)
+	alertExp := intstr.FromString(
+		fmt.Sprintf("absent(cro_aws_rds_available{namespace='%s',instanceID='%s',clusterID='%s',resourceID='%s'} == 1)",
+			cr.Namespace, instanceID, clusterID, cr.Name),
+	)
+
+	pr, err := croResources.CreatePrometheusRule(cr.Namespace, alertRuleName, alertExp)
+	if err != nil {
+		return err
+	}
+
+	// Unless it already exists, call the kubernetes api and create this PrometheusRule
+	// Replace this with CreateOrUpdate if we can figure it out
+	err = p.Client.Create(ctx, pr)
+	if err != nil {
+		if !kerrors.IsAlreadyExists(err) {
+			return errorUtil.Wrap(err, fmt.Sprintf("exception calling Create prometheusrule: %s", alertRuleName))
+		}
+	}
+	p.Logger.Info(fmt.Sprintf("PrometheusRule: %s reconcilced successfully.", pr.Name))
+	return nil
+}
+
+// DeleteRDSAvailabilityAlert We call this when we delete an ElastiCache instance,
+// it removes the prometheusrule alert which watches for the availability of the instance.
+func (p *PostgresProvider) DeleteRDSAvailabilityAlert(ctx context.Context, namespace string, instanceID string) error {
+	// query the kubernetes api to find the object we're looking for
+	alertRuleName := fmt.Sprintf("cro-aws-rds-%s", instanceID)
```

Comment:
Yeah this would be the ideal.. will take a look in the morning and see if we can make this more generic, and then move it alongside the metrics exposing functions..